### PR TITLE
Added functions to util package for new tagging feature.

### PR DESCRIPTION
**Pull request by felixtran39**
**Related**: https://github.com/GoogleContainerTools/skaffold/issues/4371

**Description**
This PR contains additional functions for the util package that will be used by the new tagging feature mentioned in #4371 . This code is only exercised in tests.

**Follow-up Work (remove if N/A)**
Next work will be the implementation of the new Tagger and its appropriate tests.

### CODE REVIEW

**Comment by gsquared94** [comment]
I suggest moving these functions [here](https://github.com/GoogleContainerTools/skaffold/blob/master/pkg/skaffold/build/tag/util.go)

**Comment by felixtran39** [comment]
@gsquared94 How come? EnvTemplate has similar functions and they are also located in the util package.

**Comment by gsquared94** [comment]
> @gsquared94 How come? EnvTemplate has similar functions and they are also located in the util package.

yes, but TagTemplate seems specific to tag creation whereas EnvTemplate can be used anywhere environment variable substitution needs to happen in a template.

**Comment by nkubala** [comment]
> yes, but TagTemplate seems specific to tag creation whereas EnvTemplate can be used anywhere environment variable substitution needs to happen in a template.

indeed, some of this code is reused for our env expansion in the custom builder, so it needed to live a level up. +1 to @gsquared94's suggestion of putting this in `pkg/skaffold/build/tag`

**Comment by felixtran39** [comment]
@nkubala @gsquared94 That makes sense. Thanks for the clarification!

File: pkg/skaffold/build/tag/tag_template.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+package tag
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/sirupsen/logrus"
+)
+
```
Comment (yuwenma) [inline pkg/skaffold/build/tag/tag_template.go:26]: a?

**Pull request merged**